### ISSUE TITLE
cd fails for directory without read permission, displays `Permission denied`

### ISSUE THREAD

**Issue by TheJonny** [fish-shell/fish-shell#10432]
`cd` or `pushd` into a directory without read permission (but with x permission) does not work in fish.

```
jonny@charon /> sh -c 'env HOME=$(mktemp -d) XDG_CONFIG_HOME= fish'
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
jonny@charon /> echo $version
3.7.1
jonny@charon /> ls -ld /srv
drwxr-x--x 1 root root 48 14. Jan 23:51 /srv/
jonny@charon /> cd /srv
cd: Permission denied: '/srv'
jonny@charon / [1]> bash
[jonny@charon /]$ cd /srv
[jonny@charon srv]$ pwd
/srv

```
This transscript was on archlinux. The  same behaviour on debian stable with fish `3.6.0`

it is possible to cd through the directory, if I have read permission in the target, e.g. `cd /srv/www`.

**Comment by TheJonny** [comment]
here an strace.

[strace.log](https://github.com/fish-shell/fish-shell/files/14917869/strace.log)

the only EACCES occurs in line 2947, followed by printing the error message:
```
2947 openat(AT_FDCWD, "/srv", O_RDONLY|O_CLOEXEC) = -1 EACCES (Permission denied)
2948 write(2, "cd: Permission denied: '/srv'\n", 30) = 30
```

**Comment by TheJonny** [comment]
`builtins/cd.rs` calls `fds::wopen_dir`, which calls `fds::open_dir`, which sets `O_RDONLY`. changing it to O_PATH works for me on Linux, but is neither portable nor correct for a function called `open_dir`

**Comment by faho** [comment]
Fixed by merging #10442.
